src/ServiceControl/Recoverability/Retrying/Infrastructure/ReturnToSenderDequeuer.cs

```diff
@@ -142,9 +142,25 @@ public virtual async Task Run(string forwardingBatchId, Predicate<MessageContext
                     Log.DebugFormat($"Waiting for forwarder for batch {forwardingBatchId} to finish.");
                 }
 
-                await syncEvent.Task;
-                registration?.Dispose();
-                await messageReceiver.StopReceive(cancellationToken);
+                try
+                {
+                    Log.Debug("Awaiting syncEvent.Task.");
+                    _ = await syncEvent.Task;
+                    Log.Debug("Awaiting syncEvent.Task.");
+
+                    Log.Debug("Disposing registration.");
+                    registration?.Dispose();
+                    Log.Debug("Registration disposed");
+
+                    Log.Debug("Stopping message receiver.");
+                    await messageReceiver.StopReceive(cancellationToken);
+                    Log.Debug("Message receiver stopped.");
+                }
+                catch (Exception ex)
+                {
+                    Log.Error("An error occurred while stopping the transport receiver.", ex);
+                    throw;
+                }
```

Comment:
I would remove the try/catch.

I also added some suggestions for the debug messages.

```suggestion
               Log.Debug("Awaiting syncEvent.Task");
               _ = await syncEvent.Task;
               Log.Debug("syncEvent.Task completed");

               Log.Debug("Disposing registration.");
               registration?.Dispose();
               Log.Debug("Registration disposed");

               Log.Debug("Stopping message receiver");
               await messageReceiver.StopReceive(cancellationToken);
               Log.Debug("Message receiver stopped");     
```